src/Sylius/Bundle/UserBundle/Resources/config/services.xml

```diff
@@ -24,7 +24,10 @@
         <parameter key="sylius.user.canonicalizer.class">Sylius\Component\User\Canonicalizer\Canonicalizer</parameter>
         <parameter key="sylius.user.password_encoder.class">Sylius\Bundle\UserBundle\Security\UserPasswordEncoder</parameter>
         <parameter key="sylius.user.password_updater.class">Sylius\Component\User\Security\PasswordUpdater</parameter>
-        <parameter key="sylius.user.code_provider.class">Sylius\Component\User\Security\TokenProvider</parameter>
+
+        <parameter key="sylius.user.token_provider.abstract_unique.class">Sylius\Component\User\Security\Provider\AbstractUniqueTokenProvider</parameter>
+        <parameter key="sylius.user.token_provider.password_reset.class">Sylius\Component\User\Security\Provider\PasswordResetTokenProvider</parameter>
```

Comment:
But it seems to me to be done not in a good way. If we would have one class, `UniqueTokenProvider` and inject to it to classes token generator, and instance of uniqueness checker it would be nicer. Then, the uniqueness checker could have one class, but be declared as few services, if we would add a repository and table name in a constructor.